tests/TestCase/Datasource/ConnectionManagerTest.php

```diff
@@ -306,6 +306,31 @@ public function dsnProvider()
                     'log' => '1'
                 ]
             ],
+            'no password' => [
+                'mysql://user@localhost:3306/database',
```

Comment:
No. If some password has been already set in the configuration, this DSN doesn't overwrite it.